### PR TITLE
Student registration

### PR DESCRIPTION
**Pull request by AmanSharma4419**
Created Back-end Auth Api

### CODE REVIEW

File: Controllers/AdminControllers.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
+// Requring The Auth
+var auth = require('../Utils/Auth');
+
+// Requring The
+var User = require('../Models/userSchema');
+
+// Admin Login Middleware
+module.exports.adminLogin = (req, res, err) => {
```
Comment (prank7) [inline Controllers/AdminControllers.js]: Don't export functions individually if there are more than one functions. Do 

`module.exports = {
  functionName: fn
}`

File: Controllers/UsersControllers.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+// Requring The UserSchema
+const User = require('../Models/userSchema');
+
+// Users Registration
+module.exports.CreatingUsers = (req,res,next) => {
```
Comment (prank7) [inline Controllers/UsersControllers.js]: Names shouldn't start with capital letter. It should be `createUser`.

File: Models/challengeSchema.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+//Creating content schema
+
+var challengeSchema = new Schema({ 
+    userId: String, 
+    contentId: String, 
+    hasSubmitted: Boolean, 
+    Submission: String, 
+    pointsAwarded: String, 
+    Type: {
```
Comment (prank7) [inline Models/challengeSchema.js]: It should be `type` and not `Type`

**Review by prank7 — CHANGES_REQUESTED** [review]

File: Controllers/AdminControllers.js (hunk 1 above)
Comment (AmanSharma4419) [reply Controllers/AdminControllers.js]: Resolved

File: Controllers/UsersControllers.js (hunk 2 above)
Comment (AmanSharma4419) [reply Controllers/UsersControllers.js]: Resolved

File: Models/challengeSchema.js (hunk 3 above)
Comment (AmanSharma4419) [reply Models/challengeSchema.js]: Resolved

File: Utils/Auth.js [hunk 4]
```diff
@@ -19,3 +19,5 @@ module.exports.verifyAdminToken = function(req, res, next) {
 		return res.json({ Token: 'Token Not Found' });
 	}
 };
+
+module.exports = {genrateToken,verifyAdminToken}
```
Comment (prank7) [inline Utils/Auth.js]: Should be spaced. Spelling is not correct.

1. Correct spelling of `generateToken`
2. rename folder from `Auth` to `auth`. `Models` to `models`

module.exports = { generateToken, verifyAdminToken }

**Review by prank7 — CHANGES_REQUESTED** [review]

File: Utils/Auth.js (hunk 4 above)
Comment (AmanSharma4419) [reply Utils/Auth.js]: Resolved

**Comment by AmanSharma4419** [comment]
@prank7 Resolved Conflicts

**Pull request merged**